### ISSUE TITLE
fix(inference): non-finite/zero temperature corrupts sampling (NaN/inf inv_temp)

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#295]
> _Issue authored by Claude (Anthropic agent) on behalf of @ohdearquant._

## Summary
A non-finite or non-positive `temperature` in `SamplingConfig` silently corrupts sampling: `1.0 / temperature` becomes `NaN` (for `temperature = NaN`) or `+inf` (for `temperature = 0.0`), and that scale factor flows into the fused top-k path, collapsing the candidate set to deterministic low token IDs regardless of the true logits. No error, no clamp — just wrong output.

## Root cause
`crates/inference/src/sampling.rs:309`:
```rust
let inv_temp = if self.config.temperature != 1.0 {
    1.0 / self.config.temperature
} else {
    1.0
};
```
`NaN != 1.0` is `true`, so `inv_temp = 1.0 / NaN = NaN`. The NaN scale propagates through `select_top_k` (sampling.rs:~521), where NaN-scaled logits are treated as `-inf` candidates, so a high-logit token can be discarded and a low token ID returned. `temperature = 0.0` yields `+inf` with similar collapse. (Conventionally `temperature = 0` means greedy/argmax, which this does NOT implement.)

## Reachability (TBV-verified at source)
```rust
use lattice_inference::sampling::{Sampler, SamplingConfig};
let cfg = SamplingConfig { temperature: f32::NAN, top_k: 2, top_p: 1.0, repetition_penalty: 1.0 };
let mut sampler = Sampler::new(cfg).with_seed(1);
let token = sampler.sample(&[0.0, 100.0, 99.0]); // can return 0 despite token 1 dominating
```

## Proposed bounded fix (1 file)
At the sampling boundary, map non-finite or non-positive `temperature` to well-defined behavior before computing `inv_temp`:
- `temperature <= 0.0` or `temperature.is_nan()` → greedy (argmax) is the conventional and least-surprising choice; OR
- reject as `InferenceError::InvalidInput` at `Sampler::new` if the config carries caller intent.

Recommend greedy-on-`temperature<=0` (matches HF/llama.cpp convention) + treat NaN/inf as `1.0` (unscaled) defensively. Add regression tests for `NaN`, `+inf`, `-inf`, and `0.0`.

## Severity
P2 — reachable from the public `Sampler` API but only with a pathological/misconfigured temperature; produces wrong output rather than a crash. Bounded fix; the only judgement call is the temperature=0 semantics (greedy vs error).

Found by an automated discovery audit; verified at source before filing.
